src/scripts/main.js

```diff
@@ -1,3 +1,251 @@
+      const tdNew = document.createElement('td');
+
+      tdNew.textContent = document.getElementById(cols[q]).value;
+
+      if (cols[q] === 'salary') {
+        tdNew.textContent = formatSalary(document.getElementById(cols[q]).value
+        );
+      }
+
+      tr.appendChild(tdNew);
+    }
+
+    document.querySelector('tbody').appendChild(tr);
+  });
+
+function formatSalary(salary) {
+  return '$' + new Intl.NumberFormat('en-GB').format(salary);
+}
+
+const form = document.querySelector('.new-employee-form');
+
+const nameEmployee = document.getElementById('name');
+
+const age = document.getElementById('age');
+const notification = document.createElement('div');
+
+notification.setAttribute('data-qa', 'notification');
+
+nameEmployee.addEventListener('input', function() {
+  if (nameEmployee.validity.valid) {
+    notification.textContent = '';
+    notification.className = 'notification error';
+  } else {
+    showError();
+  }
+});
+
+age.addEventListener('input', function() {
+  if (age.validity.valid) {
+    notification.textContent = '';
+    notification.className = 'notification';
+  } else {
+    showError();
+    notification.className = 'notification error';
+  }
+});
+
+form.addEventListener('submit', function() {
+  if (!nameEmployee.validity.valid) {
+    showError();
+    event.preventDefault();
+    notification.className = 'notification error';
+  }
+
+  if (!age.validity.valid) {
+    showError();
+    event.preventDefault();
+    notification.className = 'notification error';
+  }
+});
+
+function showError() {
+  if (nameEmployee.validity.valueMissing) {
+    notification.textContent = 'you need enter a name';
+  } else if (nameEmployee.validity.tooShort) {
+    notification.textContent
+      = `Name to short`;
+  }
+
+  if (age.validity.rangeUnderflow) {
+    notification.textContent = 'you have to be adult';
+  } else if (age.validity.rangeOverflow) {
+    notification.textContent = 'you have to be younger';
+  }
+  document.querySelector('table').appendChild(notification);
+}
+
+let area = null;
+const view = document.querySelector('table');
+
+view.addEventListener('dblclick',
+  function() {
+    event.preventDefault();
+
+    const td = event.target.closest('td');
+
+    editStart(td);
+  });
+
+function editStart(td) {
+  if (td.cellIndex === 2) {
+    area = document.createElement('select');
+
+    area.innerHTML = `<option value="Tokyo">Tokyo</option>
+    <option value="Singapore">Singapore</option>
+    <option value="London">London</option>
+    <option value="New York">New York</option>
+    <option value="Edinburgh">Edinburgh</option>
+    <option value="San Francisco">San Francisco</option>`;
+  } else {
+    area = document.createElement('input');
+    area.value = '';
+  }
+  area.className = 'cell-input';
+
+  area.onkeydown = function() {
+    if (event.key === 'Enter') {
+      this.blur();
+    }
+  };
+
+  area.onblur = function() {
+    editEnd(td);
+  };
+
+  td.replaceWith(area);
```

Comment:
don't do replace
append input inside `td` tag instead of text, but save text before it
initial value of input should be saved (initiall) text